plugins/node/opentelemetry-instrumentation-express/test/express.test.ts

```diff
@@ -493,4 +494,44 @@ describe('ExpressInstrumentation', () => {
       );
     });
   });
+  it('should work with ESM usage', async () => {
+    await testUtils.runTestFixture({
+      cwd: __dirname,
+      argv: ['fixtures/use-express.mjs'],
+      env: {
+        NODE_OPTIONS:
+          '--experimental-loader=@opentelemetry/instrumentation/hook.mjs',
+        NODE_NO_WARNINGS: '1',
+      },
+      checkResult: (err, stdout, stderr) => {
+        assert.ifError(err);
+      },
+      checkCollector: (collector: testUtils.TestCollector) => {
+        // use-express.mjs creates an express app with a 'GET /post/:id' endpoint and
+        // a `simpleMiddleware`, then makes a single 'GET /post/0' request. We
+        // expect to see spans like this:
+        //    span 'GET /post/:id'
+        //     `- span 'middleware - query'
+        //     `- span 'middleware - expressInit'
+        //     `- span 'middleware - simpleMiddleware'
+        //     `- span 'router - /post/:id'
+        const spans = collector.sortedSpans;
+        assert.strictEqual(spans[0].name, 'GET /post/:id');
+        assert.strictEqual(spans[0].kind, SpanKind.CLIENT);
+        assert.strictEqual(spans[1].name, 'middleware - query');
+        assert.strictEqual(spans[1].kind, SpanKind.SERVER);
+        assert.strictEqual(spans[1].parentSpanId, spans[0].spanId);
+        assert.strictEqual(spans[2].name, 'middleware - expressInit');
+        assert.strictEqual(spans[2].kind, SpanKind.SERVER);
+        assert.strictEqual(spans[2].parentSpanId, spans[0].spanId);
+        assert.strictEqual(spans[3].name, 'middleware - simpleMiddleware');
+        assert.strictEqual(spans[3].kind, SpanKind.SERVER);
+        assert.strictEqual(spans[3].parentSpanId, spans[0].spanId);
+        // assert.strictEqual(spans[4].name, 'router - /post/:id'); // TODO bug?
```

Comment:
Yah, I think this is a bug, or limitation. I don't think the current express instrumentation handles usage of a `express.Router()`.   Perhaps change `use-express.mjs` for now to not use a Router and we add a separate issue about handling `Router` in instr-express?

Reply:
I agree, will update

Reply:
Updated this test and created new issue for tracking #1993